File: Tests Cases/Solutions/ForwardDeclaration/ForwardDeclaration/ClassA.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ForwardDeclaration
{
    class ClassA
    {
        ClassB b;
        public ClassA()
        {
            b = new ClassB(this);
        }

        public string SayHello()
        {
            return b.SayHello();
        }
    }
}

File: Tests/003.ForwardDeclaration/Output/ClassA.cpp
#include "ClassA.h"
namespace ForwardDeclaration{

	ClassA::ClassA(){
		this->b = gc_ptr<ClassB> (new(gc)ClassB(this));
	}
	String ClassA::SayHello()
	{
		return this->b->SayHello();
	}

}